framework/include/kernels/ScalarKernel.h

```diff
@@ -39,6 +39,13 @@ class ScalarKernel : public ResidualObject, public ScalarCoupleable
    */
   virtual bool isActive() { return true; }
 
+  /**
+   * Retrieves the old value of the variable that this ScalarKernel operates on.
+   *
+   * Store this as a _reference_ in the constructor.
+   */
+  const VariableValue & uOld() const;
+
```

Comment:
Do you need this function to unbreak some particular app - or are you just assuming we'll eventually need it?

Reply:
Yeah - a few applications use the old solution.